libmscore/slur.cpp

```diff
@@ -910,76 +913,115 @@ void Slur::slurPos(SlurPos* sp)
             xo = hw1 * .5;
             if (note1)
                   yo = note1->pos().y();
-            else if(_up)
-                  yo = startCR()->bbox().top();
+            else if (_up)
+                  yo = scr->bbox().top();
             else
-                  yo = startCR()->bbox().top() + startCR()->height();
+                  yo = scr->bbox().top() + scr->height();
             yo += _spatium * .9 * __up;
 
+            // adjustments for stem and/or beam
+
             if (stem1) { //sc not null
                   Beam* beam1 = sc->beam();
                   if (beam1 && (beam1->elements().back() != sc) && (sc->up() == _up)) {
+                        // start chord is beamed but not the last chord of beam group
+                        // and slur direction is same as start chord (stem side)
+
+                        // in these cases, layout start of slur to stem
+
                         qreal sh = stem1->height() + _spatium;
                         if (_up)
                               yo = sc->downNote()->pos().y() - sh;
                         else
                               yo = sc->upNote()->pos().y() + sh;
                         xo       = stem1->pos().x();
-                        stemPos  = true;
+
+                        // force end of slur to layout to stem as well,
+                        // if start and end chords have same stem direction
+                        stemPos = true;
                         }
                   else {
+                        // start chord is not beamed or is last chord of beam group
+                        // or slur direction is opposite that of start chord
+
+                        // at this point slur is in default position relative to note on slur side
+                        // but we may need to make further adjustments
+
+                        // if stem and slur are both up
+                        // we need to clear stem horizontally
                         if (sc->up() && _up)
                               xo = hw1 + _spatium * .3;
+
                         //
                         // handle case: stem up   - stem down
                         //              stem down - stem up
                         //
                         if ((sc->up() != ecr->up()) && (sc->up() == _up)) {
-                              Note* n1  = sc->up() ? sc->downNote() : sc->upNote();
+                              // start and end chord have opposite direction
+                              // and slur direction is same as start chord
+                              // (so slur starts on stem side)
+
+                              // float the start point along the stem to follow direction of movement
+                              // see for example Gould p. 111
+
+                              // get position of note on slur side for start & end chords
+                              Note* n1  = sc->up() ? sc->upNote() : sc->downNote();
```

Comment:
Is there a particular aspect of this you are thinking isn't right?  The Finale version is different in that there aren't separate slurs for the top and bottom notes.  Actually, that;s just weird that we have a vtest like this at all, since it isn't standard notation - normally it's one slur per chord, not one per note.  So maybe that's why it looks weird.  But each of our slurs is now laid out the way Gould would have it, I think.  The basic rule we are trying to follow is, if the second note is lower than the first note, then the slur should angle down as well.  This usually happens automatically when connecting notes of the same stem direction, but when the stems have opposite directions, it's easy to encounter situations where the second note is lwoer then the first but slur angle up because it has an "up" stem and the  slur is connected to the end of the stem rather than the notehead.  We were trying to do this but getting the answer wrong in many cases.  My change just makes it work the way it seems to have been intended to work in the first place.

With the existing vtest slur-1, the significant change with my PR are the outer slurs of the bottom staff, first measure.  The top notes go E to E so the top slur should be horizontal, but the bottom notes go down from G to F, so the bottom slur should go down as well.  Again, in real life we wouldn't ever actually have two slurs; this should really have been two separate tests probably.